lib/fizzy/parser.cpp

```diff
@@ -292,9 +292,14 @@ Module parse(bytes_view input)
     input.remove_prefix(wasm_prefix.size());
 
     Module module;
+    SectionId last_id = SectionId::custom;
     for (auto it = input.begin(); it != input.end();)
     {
         const auto id = static_cast<SectionId>(*it++);
+        if (id != SectionId::custom && id <= last_id)
+            throw parser_error{"Unexpected section type"};
+        last_id = id;
```

Comment:
I think you should do this assignment only if current section is not custom

Reply:
> I think you should do this assignment only if current section is not custom

Yes, good point.